src/tally_filter.F90

```diff
@@ -946,29 +841,23 @@ end function text_label_cellborn
 !===============================================================================
 ! SurfaceFilter methods
 !===============================================================================
-  subroutine get_next_bin_surface(this, p, estimator, current_bin, next_bin, &
-       weight)
+  subroutine get_all_bins_surface(this, p, estimator, match)
     class(SurfaceFilter), intent(in)  :: this
     type(Particle),       intent(in)  :: p
     integer,              intent(in)  :: estimator
-    integer, value,       intent(in)  :: current_bin
-    integer,              intent(out) :: next_bin
-    real(8),              intent(out) :: weight
+    type(TallyFilterMatch),    intent(inout) :: match
 
     integer :: i
 
-    next_bin = NO_BIN_FOUND
-    weight = ERROR_REAL
-    if (current_bin == NO_BIN_FOUND) then
       do i = 1, this % n_bins
-        if (abs(p % surface) == this % surfaces(i)) then
-          next_bin = i
-          weight = sign(1, p % surface)
+        if (p % surface == this % surfaces(i)) then
+          call match % bins % push_back(i)
+          call match % weights % push_back(ONE)
```

Comment:
I'd suggest changing this here to:
```Fortran
if (abs(p % surface) == this % surfaces(i)) then
  call match % bins % push_back(i)
  if (p % surface < 0) then
    call match % weights % push_back(-ONE)
  else
    call match % weights % push_back(ONE)
  end if
  exit
end if
```
Pull request #893 changed the behavior for surface filters (hence why the `abs()` is now needed when comparing to the specified surfaces).